src/layers/models/layersManager.ts

```diff
@@ -43,79 +42,64 @@ class LayersManager {
     return requestedLayer;
   }
 
-  public async addLayer(layerRequest: ILayerPostRequest): Promise<void> {
-    const editJson = (jsonDocument: IMapProxyJsonDocument): IMapProxyJsonDocument => {
-      this.addNewCache(jsonDocument, layerRequest);
-
-      return jsonDocument;
-    };
-
-    await this.configProvider.updateJson(editJson);
-  }
+  public async getCacheName(layerName: string, cacheType: string): Promise<ICacheName> {
+    const errorMsg = `Layer ${layerName} or fitting cache do not exist`
+    if (!this.isCacheTypeValid(cacheType)) {
+      const invalidCacheTypeMsg = `Invalid cache type: ${cacheType} has been provided , available values: "geopackage", "s3", "file", "redis"`
+      throw new Error(invalidCacheTypeMsg);
+    }
+    const configJson = await this.configProvider.getJson();
+    const requestedLayerIndex = configJson.layers.findIndex((layer) => layer.name === layerName)
+    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+    if (requestedLayerIndex != -1) {
+      const sourcesOfLayer: string[] = configJson.layers[requestedLayerIndex].sources;
 
-  public async addLayerToMosaic(mosaicName: string, layerToMosaicRequest: ILayerToMosaicRequest): Promise<void> {
-    this.logger.info(`Add layer: ${layerToMosaicRequest.layerName} to mosaic: ${mosaicName} request`);
 
-    const editJson = (jsonDocument: IMapProxyJsonDocument): IMapProxyJsonDocument => {
-      if (!isLayerNameExists(jsonDocument, layerToMosaicRequest.layerName)) {
-        throw new NotFoundError(`Layer name '${layerToMosaicRequest.layerName}' is not exists`);
-      }
+      const requestedCache = sourcesOfLayer.find(sourceName => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const currentSourceCache: IMapProxyCache = configJson.caches[sourceName];
+        return (currentSourceCache.cache.type == cacheType)
+      });
 
-      if (!isLayerNameExists(jsonDocument, mosaicName)) {
-        throw new NotFoundError(`Mosaic name '${mosaicName}' is not exists`);
+      if (requestedCache != null) {
+        return { cacheName: requestedCache };
       }
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-      const mosaicCache: IMapProxyCache = jsonDocument.caches[mosaicName];
-      mosaicCache.sources.push(layerToMosaicRequest.layerName);
-      return jsonDocument;
-    };
-
-    await this.configProvider.updateJson(editJson);
-    this.logger.info(`Successfully added layer: '${layerToMosaicRequest.layerName}' to mosaic: '${mosaicName}'`);
+    }
+    throw new NotFoundError(errorMsg)
```

Comment:
log warn with the not found error and request params

Reply:
done